superset-frontend/src/dashboard/components/gridComponents/Chart.jsx

```diff
@@ -154,6 +157,10 @@ export default class Chart extends React.Component {
       return true;
     }
 
+    if (!isEqual(nextProps.chart, this.props.chart)) {
```

Comment:
is this different than `if (!areObjectsEqual(nextProps[prop], this.props[prop]))`?
If you don't filter the chart prop to `SHOULD_UPDATE_ON_PROP_CHANGES`, shouldn't the code reach the `areObjectEquals`?

Reply:
@diegomedina248 
I think that `isEqual` works similar to `areObjectsEqual` and `areObjectsEqual` maybe more strict in this case.

And then, when click `refresh dashboard` after change the dataset, all chart components should update whether if it is now visible or not. Thus, I define the filtering by chart props outside of `isComponentVisible` props filtering and so `chart` prop doesn't need to reach the `areObjectEquals` in `SHOULD_UPDATE_ON_PROP_CHANGES `.

Reply:
mm, but I think `isComponentVisible` might be broader, and put in place for a reason.
If we add this condition here, we might be triggering more rerenders than expected.
@rusackas what do you think?

Reply:
It makes sense to not render the chart when it is not visible, which improves performance. The same is true for fine-grained control of chart updates via SHOULD_UPDATE_ON_PROP_CHANGES.

In this issue, in the case of a chart that is not updated, I suggest finding a prop that is not updated case by case and add it to SHOULD_UPDATE_ON_PROP_CHANGES.